buildSrc/src/main/groovy/airbyte-bulk-connector.gradle

```diff
@@ -134,48 +142,87 @@ class AirbyteBulkConnectorPlugin implements Plugin<Project> {
-                resources {
-                    srcDir '.'
-                    include 'metadata.yaml'
+                // This source set should only be used for tests based on the old CDK's test classes,
+                // in particular DestinationAcceptanceTest / BaseTypingDedupingTest.
+                integrationTestLegacy {
+                    kotlin {
+                        srcDir 'src/test-integration-legacy/kotlin'
+                    }
+                    resources {
+                        srcDir 'src/test-integration-legacy/resources'
+                    }
                 }
             }
+
+            // This task exists solely so that we can run the "modern" integration tests
+            // in non-docker mode on our laptops.
+            // It's not a dependency of any other task.
+            project.tasks.register('integrationTestNonDocker', Test) {
+                description = 'Runs the integration tests in non-docker mode.'
+                group = 'verification'
+                testClassesDirs = project.sourceSets.integrationTest.output.classesDirs
+                classpath = project.sourceSets.integrationTest.runtimeClasspath
+                useJUnitPlatform()
+            }
+
+            // For historical reasons (i.e. airbyte-ci), this task is called integrationTestJava.
+            // airbyte-ci only wants to run a task with exactly that name.
+            // There's intentionally no equivalent to run integrationTestNonDocker + integrationTestLegacy.
+            // It's an actual Test task (rather than just depending on two sub-tasks integrationTestDocker + integrationTestLegacy)
+            // so that gradle args (e.g. `--tests`) work as expected.
+            project.tasks.register('integrationTestJava', Test) {
+                description = 'Runs the integration tests in docker mode.'
+                group = 'verification'
+                testClassesDirs = project.sourceSets.integrationTest.output.classesDirs + project.sourceSets.integrationTestLegacy.output.classesDirs
+                classpath = project.sourceSets.integrationTest.runtimeClasspath + project.sourceSets.integrationTestLegacy.runtimeClasspath
+                useJUnitPlatform()
+                // We need a docker image to run this task, so depend on assemble
+                dependsOn project.tasks.assemble
+                environment "MICRONAUT_ENVIRONMENTS", "DOCKERIZED_INTEGRATION_TEST"
+            }
+
+            project.dependencies {
+                // integration tests depend on the runtime code
+                integrationTestImplementation project
+                integrationTestLegacyImplementation project
+            }
+
+            project.tasks.named('build').configure {
+                dependsOn project.tasks.integrationTestJava
+            }
         }
-        project.tasks.matching {it.name == 'spotbugsIntegrationTestJava' || it.name == 'spotbugsTest' }.configureEach {
+
+        project.tasks.matching {
```

Comment:
Does this definitely keep spot bugs from running? I tried a few versions of this earlier and couldn't get it to work.

Reply:
yep - tried it out locally, I don't get a `build/reports/spotbugs` directory unless I flip this to `enabled = true`